src/wp-includes/modules.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Modules API: Module functions
+ *
+ * @since 6.5.0
+ *
+ * @package WordPress
+ * @subpackage Modules
+ */
+
+/**
+ * Retrieves the main WP_Modules instance.
+ *
+ * This function provides access to the WP_Modules instance, creating one if it
+ * doesn't exist yet.
+ *
+ * @since 6.5.0
+ *
+ * @return WP_Modules The main WP_Modules instance.
+ */
+function wp_modules() {
+	static $instance = null;
+	if ( is_null( $instance ) ) {
+		$instance = new WP_Modules();
+	}
+	return $instance;
+}
+
+/**
+ * Registers the module if no module with that module identifier has already
+ * been registered.
+ *
+ * @since 6.5.0
+ *
+ * @param string            $module_identifier The identifier of the module. Should be unique. It will be used in the
+ *                                             final import map.
+ * @param string            $src               Full URL of the module, or path of the script relative to the WordPress
+ *                                             root directory.
+ * @param array             $dependencies      Optional. An array of module identifiers of the dependencies of this
```

Comment:
See note above about possibly incorporating a more strict PHP type with an array shape.

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/5818/commits/df73485c9c43995eea8b298138df43e258cef0d4.